testutil/integration/evmos/utils/bank_test.go

```diff
@@ -0,0 +1,65 @@
+package utils_test
+
+import (
+	"testing"
+
+	"cosmossdk.io/math"
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	banktypes "github.com/cosmos/cosmos-sdk/x/bank/types"
+	grpchandler "github.com/evmos/evmos/v16/testutil/integration/evmos/grpc"
+	testkeyring "github.com/evmos/evmos/v16/testutil/integration/evmos/keyring"
+	"github.com/evmos/evmos/v16/testutil/integration/evmos/network"
+	"github.com/evmos/evmos/v16/testutil/integration/evmos/utils"
+	"github.com/stretchr/testify/require"
+)
+
+func TestCheckBalances(t *testing.T) {
+	testDenom := "atest"
+	keyring := testkeyring.New(1)
+	nw := network.New(
+		network.WithDenom(testDenom),
+		network.WithPreFundedAccounts(keyring.GetAllAccAddrs()...),
+	)
+	handler := grpchandler.NewIntegrationHandler(nw)
+
+	testcases := []struct {
+		name        string
+		address     string
+		expAmount   math.Int
+		expPass     bool
+		errContains string
+	}{
+		{
+			name:      "pass",
+			address:   keyring.GetAccAddr(0).String(),
+			expAmount: network.PrefundedAccountInitialBalance,
+			expPass:   true,
+		},
+		{
+			name:        "fail - wrong amount",
+			address:     keyring.GetAccAddr(0).String(),
+			expAmount:   sdk.NewInt(1),
+			errContains: "expected balance",
+		},
+	}
+
+	for _, tc := range testcases {
+		tc := tc
```

Comment:
Why is this needed?

Reply:
This is a super weird pattern in Go I've noticed 👀 in the past we've had problems where the `tc` fields were only using values of the last testcase - this solves it. It's very weird though and didn't occur on all tests though .. you'll also find this pattern everywhere in the Cosmos SDK repo for example and also we have it in our codebase a lot

<img width="300" alt="image" src="https://github.com/evmos/evmos/assets/42640438/bb21cd7a-8dab-41cd-99b2-d4e2a3bdb7d4">